core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemListener.java

```diff
@@ -166,39 +168,40 @@ public FS getFileSystem() {
 	public synchronized F getRawMessage(Map<String,Object> threadContext) throws ListenerException {
 		try {
 			FS fileSystem=getFileSystem();
-			Iterator<F> it = fileSystem.listFiles(getInputFolder());
-			if (it==null || !it.hasNext()) {
-				return null;
-			}
-			
-			long stabilityLimit = getMinStableTime();
-			if (stabilityLimit>0) {
-				stabilityLimit=System.currentTimeMillis()-stabilityLimit;
-			}
-			while (it.hasNext()) {
-				F file = it.next();
+			try(DirectoryStream<F> ds = fileSystem.listFiles(getInputFolder())) {
+				Iterator<F> it = ds.iterator();
```

Comment:
These two try statements can be combined. The getFileSystem() need not be inside the try.